### ISSUE TITLE
Test building of projects generated from templates

### ISSUE THREAD

**Issue by ccheetham** [SteeltoeOSS/NetCoreToolTemplates#31]
Implement a test matrix with dimensions:
* Steeltoe version
* .NET target framework
* option

For example, a point in the matrix would test:
```
$ dotnet new stwebapi --framework netcoreapp3.1 --steeltoe 3.0.2 --dynamic-logging
```

To pass, a project should succeed in:
* `dotnet restore`
* `dotnet build /p:TreatWarningsAsErrors=True`

**Comment by ccheetham** [comment]
Implemented per d106e00

**Comment by jkonicki** [comment]
Only tests locally not via CI builds
